src/components/ViewPDF.tsx

```diff
@@ -0,0 +1,139 @@
+import { Box, Stack, useTheme } from "@mui/material";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import LaunchIcon from "@mui/icons-material/Launch";
+import PreviewIcon from "@mui/icons-material/PictureInPictureAlt";
+import DownloadIcon from "@mui/icons-material/FileDownload";
+import { IconMessage, Modal, SolidOutline } from "components";
+import { FunctionComponent, useRef, useState } from "react";
+
+interface ViewPdfProps {
+  /** True if the PDF has been viewed */
+  readonly isViewed?: boolean;
+  /** Called when the agreement is previewed or downloaded */
+  readonly onViewPDF?: VoidFunction;
+  /** Encoded base64 string of PDF file */
+  readonly data: string;
+  /** Preview image for PDF file */
+  readonly preview: string;
+}
+
+/**
+ * Component to preview or download a PDF file.
+ */
+const ViewPDF: FunctionComponent<ViewPdfProps> = ({
+  data,
+  preview,
+  onViewPDF,
+  isViewed = false,
+}) => {
+  const linkRef = useRef<HTMLAnchorElement | null>(null);
+
+  const { colors } = useTheme();
+
+  const [isHovered, setIsHovering] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const agreementData = `data:application/pdf;base64,${data}`;
+
+  const handlePreview = () => {
+    setIsModalOpen(true);
+    if (onViewPDF) onViewPDF();
+  };
+
+  const handleDownload = () => {
+    linkRef.current?.click();
+    if (onViewPDF) onViewPDF();
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+  };
+
+  return (
+    <>
+      <Box>
+        <a
+          ref={ linkRef }
+          style={ { display: "none" } }
+          href={ agreementData }
+          download="artist-agreement.pdf"
+        >
+          Download
+        </a>
```

Comment:
Was this just a test link or is this a fall back?

Reply:
Nevermind, didn't understand why the Download text for this wasn't visible, but understand this is for the link to the proposed document.